pkg/controller/gitopsservice/gitopsservice_controller.go

```diff
@@ -493,6 +544,10 @@ func (r *ReconcileGitopsService) reconcileBackend(gitopsserviceNamespacedName ty
 				found.Spec.Template.Spec.Containers[0].Args = deploymentObj.Spec.Template.Spec.Containers[0].Args
 				changed = true
 			}
+			if found.Spec.Template.Spec.Containers[0].Resources.Requests == nil {
+				found.Spec.Template.Spec.Containers[0].Resources = deploymentObj.Spec.Template.Spec.Containers[0].Resources
+				changed = true
+			}
```

Comment:
We probably should do something like this instead of checking for non nil `Requests` field.
```
if !reflect.DeepEqual(deploy.Spec.Template.Spec.Containers[0].Resources, deploymentObj.Spec.Template.Spec.Containers[0].Resources) {

```

Reply:
acknowledge :)